### PR TITLE
322 compact messages

### PR DESCRIPTION
**Pull request by Natasha08**
### Checklist:
- [x] Test your work and double check you didn't break anything

### Description:
[Issue#322](https://github.com/OriginProtocol/origin-dapp/issues/322)

Currently, every time a user sends a message the content is displayed with the avatar, name & timestamp resulting in repetitive visual clutter. 

If the latest message is from the same sender as the previous message and the time between these messages is less than 10 minutes, then we only display the message content. Otherwise we display the message content with the other information. I created a separate component to handle compacting the messages.
￼
image

### CODE REVIEW

**Comment by micahalcorn** [comment]
Thanks, @Natasha08 🙏 I'll take a look now.

File: src/css/app.css [hunk 1]
```diff
@@ -2482,6 +2482,10 @@ form .stars img {
   margin-bottom: 20px;
 }
 
+.conversation .compact-message {
+  margin: -20px auto 20px 25px
```
Comment (micahalcorn) [inline src/css/app.css]: Missing a ; 😁

File: src/components/message.js [hunk 2]
```diff
@@ -37,7 +37,7 @@ class Message extends Component {
               {moment(created).format('MMM Do h:mm a')}
             </div>
           </div>
-          <div className="message-content">
+          <div>
```
Comment (micahalcorn) [inline src/components/message.js]: We're now using this `message-content` class, so we'll need it back 😬

**Comment by Natasha08** [comment]
@micahalcorn let me know what you think. I'm still working on manual re-testing -- the dApp is in a weird state where I need to enable messaging, but I am not presented with the option. I'll let you know when I have manually tested with the new changes.

**Comment by micahalcorn** [comment]
As I test more, I see the reasoning for negative `margin-top`. This is tricky depending on length of the non-compact message, presence of line breaks, etc. 😐

**Comment by micahalcorn** [comment]
I've made some additional tweaks but the vertical spacing isn't perfect. I think a case could be made that we should have a logical unit for the group of closely timed messages and then actually render all of those content strings inside of a parent container - for example, all within the `<div className="message-content">`. So then instead of a Message component, we would have something like a MessageGroup component containing one or more MessageContent components. But then we would have to figure out the CSS for letting the second line of message content wrap under the timestamp. Since this UI is still in flux, I think we've done all that we should do for now. Thanks again @Natasha08 for the contribution! 👍

**Pull request merged**